src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -449,7 +445,7 @@ public virtual void WriteTo([NotNull] TextWriter writer, object value)
                     //
                     // Do not combine following condition with check of escapeQuotes; htmlString.ToString() can be
                     // expensive when the HtmlString is created with a StringCollectionTextWriter.
-                    var stringValue = htmlString.ToString();
+                    var stringValue = htmlContent.ToString();
```

Comment:
I'm not happy to see a `ToString()` - but I get why it's showing up. Let's just make a mental note to revist as part of the perf part of this change.